dspace-oai/src/main/java/org/dspace/xoai/util/DateUtils.java

```diff
@@ -42,50 +45,62 @@ public static String format(Date date, boolean init)
 
     public static Date parse(String date)
     {
-        // 2008-01-01T00:00:00Z
+        // in org.dspace.xoai.util.DateUtils.format(Date, boolean) we format all
+        // dates with trailing milliseconds so we have to use them when parsing
         SimpleDateFormat format = new SimpleDateFormat(
-                "yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault());
-        // format.setTimeZone(TimeZone.getTimeZone("ZULU"));
+                "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+        format.setTimeZone(TimeZone.getTimeZone("ZULU"));
         Date ret;
         try
         {
             ret = format.parse(date);
             return ret;
         }
-        catch (ParseException e)
+        catch (ParseException ex)
         {
-            format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss",
+            // 2008-01-01T00:00:00Z
+            format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'",
```

Comment:
The problem is that I can't tell what time zone to use for the other cases (like yyyy-MM-dd HH:mm:ss). So currently they are all set to ZULU. Should I change that to use the server time in the cases where the time zone is not clear?